crates/red_knot_python_semantic/src/types.rs

```diff
@@ -340,7 +395,9 @@ pub enum Type<'db> {
     /// General rule: `Todo` should only propagate when the presence of the input `Todo` caused the
     /// output to be unknown. An output should only be `Todo` if fixing all `Todo` inputs to be not
     /// `Todo` would change the output type.
-    Todo,
+    ///
+    /// This variant should be created with the `todo_type!` macro.
```

Comment:
I reverted this, as `cargo doc` failed. `todo_type!` is only `pub(crate)`, but `Type` is `pub`. I don't know if it would be fine to make `Type` `pub(crate)`, and it's not straightforward anyway. And I don't really want to make the macro `pub`, as that would require using `#[macro_export]` and `$crate` in both variants of the macro... which seems like a bit too much for this small QOL improvement.